x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/between/BetweenFunctionProcessor.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.eql.expression.function.scalar.between;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.eql.EqlIllegalArgumentException;
+import org.elasticsearch.xpack.ql.expression.gen.processor.Processor;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class BetweenFunctionProcessor implements Processor {
+
+    public static final String NAME = "sbetween";
+
+    private final Processor source, left, right, greedy, caseSensitive;
+
+    public BetweenFunctionProcessor(Processor source, Processor left, Processor right, Processor greedy, Processor caseSensitive) {
+        this.source = source;
+        this.left = left;
+        this.right = right;
+        this.greedy = greedy;
+        this.caseSensitive = caseSensitive;
+    }
+
+    public BetweenFunctionProcessor(StreamInput in) throws IOException {
+        source = in.readNamedWriteable(Processor.class);
+        left = in.readNamedWriteable(Processor.class);
+        right = in.readNamedWriteable(Processor.class);
+        greedy = in.readNamedWriteable(Processor.class);
+        caseSensitive = in.readNamedWriteable(Processor.class);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(source);
+        out.writeNamedWriteable(left);
+        out.writeNamedWriteable(right);
+        out.writeNamedWriteable(greedy);
+        out.writeNamedWriteable(caseSensitive);
+    }
+
+    @Override
+    public Object process(Object input) {
+        return doProcess(source.process(input), left.process(input), right.process(input),
+                greedy.process(input), caseSensitive.process(input));
+    }
+
+    public static Object doProcess(Object source, Object left, Object right, Object greedy, Object caseSensitive) {
+        if (source == null) {
```

Comment:
what if left and right are null?
```suggestion
        if (source == null || left == null || right == null) {
```